.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,38 @@
+name: Hextractor build and test
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+
+permissions:
+  contents: read
+
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: .
+    timeout-minutes: 3
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install poetry
+        run: pipx install poetry
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+          cache: "poetry"
+      - name: Install deps
+        run: poetry install --with dev
+      - name: Lint
+        run: |
+          poetry run ruff check --exit-zero
```

Comment:
Adding `--exit-zero` cause ruff check exits with status code "0", even upon detecting lint violations, but I think the build should fail in this case, so I would change this line like this:

```diff
-poetry run ruff check --exit-zero
+poetry run ruff check
```

Reply:
Fixed, thanks!